tests/unit/test_exceptions.py

```diff
@@ -235,3 +235,89 @@ def test_from_grpc_error_non_call():
     assert exception.message == message
     assert exception.errors == [error]
     assert exception.response == error
+
+
+def create_bad_request_details():
+    bad_request_details = error_details_pb2.BadRequest()
+    field_violation = bad_request_details.field_violations.add()
+    field_violation.field = "document.content"
+    field_violation.description = "Must have some text content to annotate."
+    status_detail = any_pb2.Any()
+    status_detail.Pack(bad_request_details)
+    return status_detail
+
+
+def test_error_details_from_rest_response():
+    bad_request_detail = create_bad_request_details()
+    status = status_pb2.Status()
+    status.code = 3
+    status.message = (
+        "3 INVALID_ARGUMENT: One of content, or gcs_content_uri must be set."
+    )
+    status.details.append(bad_request_detail)
+
+    # See JSON schema in https://cloud.google.com/apis/design/errors#http_mapping
+    http_response = make_response(
+        json.dumps({"error": json.loads(json_format.MessageToJson(status))}).encode(
+            "utf-8"
+        )
+    )
+    exception = exceptions.from_http_response(http_response)
+    want_error_details = [json.loads(json_format.MessageToJson(bad_request_detail))]
+    assert want_error_details == exception.details
+    # 404 POST comes from make_response.
+    assert str(exception) == (
+        "404 POST https://example.com/: 3 INVALID_ARGUMENT:"
+        " One of content, or gcs_content_uri must be set."
+        " [{'@type': 'type.googleapis.com/google.rpc.BadRequest',"
+        " 'fieldViolations': [{'field': 'document.content',"
+        " 'description': 'Must have some text content to annotate.'}]}]"
+    )
+
+
+def test_error_details_from_v1_rest_response():
+    response = make_response(
+        json.dumps(
+            {"error": {"message": "\u2019 message", "errors": ["1", "2"]}}
+        ).encode("utf-8")
+    )
+    exception = exceptions.from_http_response(response)
+    assert exception.details == []
+
+
+def test_error_details_from_grpc_response():
```

Comment:
Now that #289 has landed on `main`, we need to skip this test when `grpc` cannot be imported.
```suggestion
@pytest.mark.skipif(grpc is None, reason="gRPC not importable")
def test_error_details_from_grpc_response():
```